### ISSUE TITLE
Replacing small bags with big ones

### ISSUE THREAD

**Issue by Klala** [dalaranwow/dalaran-wow#3701]
**Description**: There is a problem with swapping bags, normally you should be able to swap small bags with bigger ones without any problems but here you get an error saying that the bag is full and you need to empty it first or something like that.

**Current behaviour**: You can't swap a small bag with a big bag, (swap a 6 slot bag with a 16 slot bag). You can do that only if the smaller bag is empty.

**Expected behaviour**: You should be able to swap a small bag with a bigger one without emptying the smaller bag first. The items should not interfere with the swapping.

**Steps to reproduce the problem**:

1. Have a 2 bags one with more slots than the other
2. Put some items in the small bag while the other bag is ready for swapping in the backpack or another bag.
3. Try to replace the smaller bag with the bigger one.

**Include proofs for this behaviour**
http://wowwiki.wikia.com/wiki/Bag

**"If you want to equip a higher-capacity bag in place of a lower-capacity one (e.g, because it's full), you can swap out a larger nested bag for the smaller one and the items will automatically be transferred to the new bag. The smaller one then becomes empty and nested."**

**Comment by Jokey123** [comment]
I don't think this was implemented in Wrath of the Lich King.

**Comment by Klala** [comment]
It did work like that, I know because I played in retail since vanilla, I played as a tailor and made bigger bags all the time for my characters and never had a problem swapping bigger bags with small ones. The error I'm getting in game right now should be when you're trying to swap a small bag with a bigger bag, thus telling you that you need to empty the bag because the smaller bag doesn't have enough room for the items.

**Comment by Jokey123** [comment]
It's just that I encountered tons of people on the battle.net forums not knowing about this, they always thought that they had to empty their bags beforehand. And the thing that lead to that thought was them actually not being able to swap the bags unless empty. But if you've truly experienced this on retail back in Vanilla, then I guess you're right.

**Comment by Klala** [comment]
https://wow.curseforge.com/addons/tbag-shefki/tickets/28-auto-swap-bags-or-auto-empty-bag/

You can see here that someone is saying what you are saying basically but the author is telling them that the mechanic is already in the game.
I am on the lookout for a more solid proof but it's hard finding a video about someone casually equipping a bigger slot bag replacing his small one.

**Comment by kalvindeane** [comment]
OP is correct.

**Comment by Saben65** [comment]
Well it behaved how OP said it did at least as early as 2012.

https://www.youtube.com/watch?v=B1-xmkHErKU

Thank you anonymous girlfriend who couldn't handle basic game mechanics.

**Comment by Klala** [comment]
I'm not sure what you're trying to say...
If you're thinking that this video proves that swapping bags started at 2012 than you'd be wrong.
I did see this video when I searched a proof but I can't give a video from 2012 as it's end of cata\mop.
And again, I have showed evidence that hints that swapping bags did work before cata.
Dalaran WoW has a lot of minor core bugs such as being interrupted while gathering\casting when you dismount from a flying mount. This bug shouldn't come as a surprise.

The only thing this video shows is how the bag swapping should work which is good for the developers I guess.

**Comment by robinsch** [comment]
https://github.com/TrinityCore/TrinityCore/pull/17620

**Comment by Klala** [comment]
I don't know why the people who work on trinitycore are saying this isn't blizzlike because it is.
But I realized I had to go and look for a better proof. So I went back and thought out of the box just to find a video that shows what I said is true.
https://www.youtube.com/watch?v=ZSoDEJxsXtg&t=14m45s

This is Joana. He was a player who made a video back in vanilla on how to speed level your character to 60. This was filmed around 2005-2006. Since there are no changes to bags mechanics in the patch logs ever I think it's safe to say that it should work in 3.3.5 too.
Notice how in 14:45 he has all his bags full with contents and suddenly he finds a new bag bigger than what he has and he replaces them without any problems. This is how it should work.

**Comment by Edzero** [comment]
OP is certainly correct in how it worked on retail during WotLK.

The reason this isn't implemented in Trinity Core as standard is because it would open up ways to cheat using the current system in Trinity. This is because of the item backreferences needing an update when swapping containers.

**Comment by Klala** [comment]
Yes but for some reason people who work on doing this blizzlike doesn't know it's blizzlike.
This issue should be reopened regardless of it being easy or hard to fix because it needs fixing and if Dalaran WoW can fix it for the good of all the private servers then it is another step for them being perfectly scripted.

**Comment by robinsch** [comment]
@Klala Your first video is from a private server.

**Comment by Edzero** [comment]
https://github.com/TrinityCore/TrinityCore/pull/17650

**Comment by Klala** [comment]
@robinsch What first video? The first wotlk reference? If so pardon me, I don't know german at all so that might have been my mistake.
Then again, some private servers have that going but they might be working with mangos or some other core. I just know that on mangos core this works.
